### PR TITLE
Extend docs on $schema

### PR DESCRIPTION
**Pull request by domoritz**
Follow up for #2933

### CODE REVIEW

File: src/spec.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import {contains, Dict, duplicate, hash, vals} from './util';
 
 export type TopLevel<S extends BaseSpec> = S & TopLevelProperties & {
   /**
-   * URL to JSON schema for this Vega-Lite specification.
+   * URL to [JSON schema](http://json-schema.org/) for this Vega-Lite specification. Unless you have a reason to change this, use `https://vega.github.io/schema/vega-lite/v2.json`. Setting the `$schema` property allows automatic validation and autocomplete in editors that support JSON schema.
```
Comment (kanitw) [inline src/spec.ts]: "for a Vega-Lite specification"?

(This => a)

Comment (domoritz) [reply src/spec.ts]: I think this is correct here as you can have your own dialect of Vega-Lite or use a specific version.

Comment (kanitw) [reply src/spec.ts]: This will appear in the docs, and it will be weird which specification is "this"?

Comment (domoritz) [reply src/spec.ts]: Done

**Pull request merged**